cpp/include/cudf/strings/detail/gather.cuh

```diff
@@ -73,7 +73,11 @@ std::unique_ptr<cudf::column> gather( strings_column_view const& strings,
       mstate = ALL_NULL;
     }
     // create null mask -- caller must update this
-    rmm::device_buffer null_mask = create_null_mask( output_count, mstate, stream, mr );
+
+    rmm::device_buffer null_mask{};
```

Comment:
This looks like a good case to use an [`IIFE`](https://www.bfilipek.com/2016/11/iife-for-complex-initialization.html) (thanks to @codereport for introducing me to these).

```suggestion
    rmm::device_buffer null_mask  = [&]{
       if(strings.parent().nullable() or NullifyOutOfBounds)
          return create_null_mask(output_count, mstate, stream, mr);
       else
          return device_buffer{};
    }();
```